### ISSUE TITLE
Aggiungere gestione multilingua

### ISSUE THREAD

**Issue by nazarenomanco** [DuoMes/duo#41]
Usiamo WPFLocalizeExtension? 
Inseriamo dal subito per non dover ripassare tutto il sorgente?

**Comment by mauroservienti** [comment]
Quale è il vantaggio di usare quello rispetto al normalissimo binding con un `ResourceDictionary`?

**Comment by nazarenomanco** [comment]
Per quello che ci serve ora forse nessuno ma è un prodotto maturo che nelle mie precedenti esperienze di applicazioni multilingue fa il suo lavoro bene e in modo poco invasivo.

https://wpflocalizeextension.codeplex.com/

Giusto per non partire con un ResourceDictionary e poi accogerci di aver bisogno di qualcosa di più.

**Comment by mauroservienti** [comment]
Vendimi che cosa fa in più rispetto ad un `Resx`, non voglio fare il polemico voglio solo capire quali sono i vantaggi rispetto a gestire con `Resx` e `DataBinding`.

**Comment by nazarenomanco** [comment]
@mauroservienti se per quanto riguarda le labels possiamo procedere con i Resx (che in futuro, se serve ci daranno possibilità di usare tools di traduzione pro), come facciamo con i valori contenuti sul DB (ad esempio la descrizione del trattamento)?

Switch a caldo o restart?

**Comment by mauroservienti** [comment]
> @mauroservienti se per quanto riguarda le labels possiamo procedere con i Resx (che in futuro, se serve ci daranno possibilità di usare tools di traduzione pro), come facciamo con i valori contenuti sul DB (ad esempio la descrizione del trattamento)?

Usiamo la `descrizione del trattamento` come esempio, mi stai dicendo che:

* la descrizione è un valore imputato dall'utente, in fase di configurazione o in fase operativa poco importa
* nella **stessa** installazione ci sono utenti che vogliono vedere il valore di cui sopra in lingue diverse

Giusto?

> Switch a caldo o restart?

se usiamo i `ResX` con data binding per la UI lo switch a caldo è compreso nel prezzo 😄

**Comment by nazarenomanco** [comment]
Si corretto... Pensavo a una soluzione traduzione su DB... Ma se troviamo come renderla uniforme tanto meglio. 
Soluzione attuale : una tabella aggiuntiva con le traduzioni nelle varie lingue supportate

**Comment by mauroservienti** [comment]
ma è un macello, significa che in fase di inserimento di un valore devi inserire tutte le lingue giusto? Ergo sto editando la `descrizione del trattamento`, le lingue supportate sono 4, devo farti vedere 4 `Textbox`. Giusto?

Non solo ammazza anche le performance, perché ogni query diventa una `join`.

* Ma di quanti tipi di valori/quanti valori stiamo parlando? Cioè se le lingue supportate sono 4 e devo inserire note per un processo di estrusione devo farlo in 4 lingue?
* Sono solo valori di configurazione? o anche altro?

**Comment by nazarenomanco** [comment]
Ci sono "le label" e quelle le facciamo con i ResX

Descrizione delle anagrafiche in lingua. In questo caso, al momento, la soluzione usata da altre parti è una sorta di left outer join un una tabella con le traduzioni (se non ci sono fallback su standard = inglese). 

Per rendere la cosa meno "left outer" siamo passati a copiare l'inglese sulla nuova lingua e poi se lo sistemano (da configurazione delle anagrafiche). 

Le anagrafiche non sono molte... ma servono in varie parti della applicazione (il trattemento viene abbinato alle bobine ad esempio)

Sono comunque tutte "tabelle" la cui necessità di traduzione è conosciuta a priori, ma il numero di righe (ad esempio il numero e la codifica dei trattamenti) è decisa dall'utente.

**Comment by mauroservienti** [comment]
e se cambiano cambiano solo cambiando la configurazione, che presumo non essere un'operazione che succede tutti i giorni. Nel qual caso potremmo cachare queste informazioni sul client. Giusto?

**Comment by nazarenomanco** [comment]
Giusto, con un modo di forzare il ricaricamento e un ricaricamento ogni x tempo ok.

Inviato dal mio dispositivo Huawei

-------- Messaggio originale --------
Oggetto: Re: [DuoMes/duo] Aggiungere gestione multilingua (#41)
Da: Mauro Servienti
A: DuoMes/duo
CC: Nazareno Manco ,Author


e se cambiano cambiano solo cambiando la configurazione, che presumo non essere un'operazione che succede tutti i giorni. Nel qual caso potremmo cachare queste informazioni sul client. Giusto?

**Comment by mauroservienti** [comment]
Avendo un socket SignalR aperto con i client, mandare un messaggio per invalidare la cache locale è banale.
